cpp/src/Ice/Incoming.cpp

```diff
@@ -1,672 +0,0 @@
-            {
-                _os.write(replyUnknownUserException);
-                _os.write(uue->unknown, false);
-            }
-            else if (const UnknownException* ue = dynamic_cast<const UnknownException*>(&ex))
-            {
-                _os.write(replyUnknownException);
-                _os.write(ue->unknown, false);
-            }
-            else if (const LocalException* le = dynamic_cast<const LocalException*>(&ex))
-            {
-                _os.write(replyUnknownLocalException);
-                ostringstream str;
-                str << *le;
-                if (IceUtilInternal::printStackTraces)
-                {
-                    str << '\n' << ex.ice_stackTrace();
-                }
-                _os.write(str.str(), false);
-            }
-            else if (const UserException* use = dynamic_cast<const UserException*>(&ex))
-            {
-                _os.write(replyUnknownUserException);
-                ostringstream str;
-                str << *use;
-                if (IceUtilInternal::printStackTraces)
-                {
-                    str << '\n' << ex.ice_stackTrace();
-                }
-                _os.write(str.str(), false);
-            }
-            else
-            {
-                _os.write(replyUnknownException);
-                ostringstream str;
-                str << ex;
-                if (IceUtilInternal::printStackTraces)
-                {
-                    str << '\n' << ex.ice_stackTrace();
-                }
-                _os.write(str.str(), false);
-            }
-
-            _observer.reply(static_cast<int32_t>(_os.b.size() - headerSize - 4));
-            _responseHandler->sendResponse(_current.requestId, &_os, _compress);
-        }
-        else
-        {
-            _responseHandler->sendNoResponse();
-        }
-    }
-    catch (...)
-    {
-        string exceptionId = getExceptionId(exc);
-
-        if (_os.instance()->initializationData().properties->getPropertyAsIntWithDefault("Ice.Warn.Dispatch", 1) > 0)
-        {
-            warning(exc);
-        }
-
-        if (_observer)
-        {
-            _observer.failed(exceptionId);
-        }
-
-        if (_isTwoWay)
-        {
-            _os.writeBlob(replyHdr, sizeof(replyHdr));
-            _os.write(_current.requestId);
-            _os.write(replyUnknownException);
-            ostringstream str;
-            str << "c++ exception: " << exceptionId;
-            _os.write(str.str(), false);
-
-            _observer.reply(static_cast<int32_t>(_os.b.size() - headerSize - 4));
-            _responseHandler->sendResponse(_current.requestId, &_os, _compress);
-        }
-        else
-        {
-            _responseHandler->sendNoResponse();
-        }
-    }
-
-    _observer.detach();
-    _responseHandler = nullptr;
-}
-
-void
-Incoming::invoke(const ServantManagerPtr& servantManager, InputStream* stream)
```

Comment:
This dispatch logic moved to the IncomingRequest constructor and ServantManager::dispatch.